File: Elitesportweb/elite_sports_flutter/lib/models/exercise_plan.dart
class ExercisePlan {
  final String id;
  final String planName;
  final String description;
  final String assignedBy;
  final String? duration;
  final String? notes;
  final List<Exercise> exercises;

  ExercisePlan({
    required this.id,
    required this.planName,
    required this.description,
    required this.assignedBy,
    this.duration,
    this.notes,
    required this.exercises,
  });

  factory ExercisePlan.fromJson(Map<String, dynamic> json) {
    return ExercisePlan(
      id: json['_id'] ?? '',
      planName: json['planName'] ?? '',
      description: json['description'] ?? '',
      assignedBy: json['assignedBy'] ?? '',
      duration: json['duration'],
      notes: json['notes'],
      exercises: (json['exercises'] as List? ?? [])
          .map((exercise) => Exercise.fromJson(exercise))
          .toList(),
    );
  }
}

class Exercise {
  final String name;
  final String sets;
  final String reps;
  final String? weight;
  final String? duration;
  final String? instructions;

  Exercise({
    required this.name,
    required this.sets,
    required this.reps,
    this.weight,
    this.duration,
    this.instructions,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      name: json['name'] ?? '',
      sets: json['sets'] ?? '',
      reps: json['reps'] ?? '',
      weight: json['weight'],
      duration: json['duration'],
      instructions: json['instructions'],
    );
  }
}

File: Elitesportweb/models/ExercisePlan.js
import mongoose from 'mongoose'

const ExercisePlanSchema = new mongoose.Schema({
  memberId: { type: mongoose.Schema.Types.ObjectId, ref: 'Member', required: true },
  instructorId: { type: mongoose.Schema.Types.ObjectId, ref: 'Instructor', required: true },
  title: { type: String, required: true },
  description: { type: String },
  exercises: [{
    name: { type: String, required: true },
    sets: { type: Number, required: true },
    reps: { type: String, required: true }, // e.g., "10-12" or "30 seconds"
    weight: { type: String }, // e.g., "20kg" or "bodyweight"
    restTime: { type: String }, // e.g., "60 seconds"
    notes: { type: String }
  }],
  category: { type: String, required: true, enum: ['crossfit', 'karate', 'zumba', 'general'] },
  difficulty: { type: String, required: true, enum: ['beginner', 'intermediate', 'advanced'] },
  duration: { type: String }, // e.g., "4 weeks"
  frequency: { type: String }, // e.g., "3 times per week"
  goals: [{ type: String }],
  status: { type: String, default: 'active', enum: ['active', 'completed', 'paused'] },
  approvalStatus: { type: String, default: 'pending', enum: ['pending', 'approved', 'rejected'] },
  approvedBy: { type: mongoose.Schema.Types.ObjectId, ref: 'Admin' },
  approvedAt: { type: Date },
  rejectionReason: { type: String }
}, { timestamps: true })

export default mongoose.models.ExercisePlan || mongoose.model('ExercisePlan', ExercisePlanSchema)